File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/typescript/controls/boundaries.ts
class Router {
  constructor(
    private primary: () => void, // @S1
    private secondary: () => void, // @S2
  ) {}
  get callback(): () => void { return this.primary; }
  set callback(f: () => void) { this.secondary = f; }
  get other(): () => void { return this.secondary; }
  fire() {
    const fn = this.callback;
    fn(); // @I1
  }
  fireOther() {
    const fn = this.other;
    fn(); // @I2
  }
  change(f: () => void) { this.callback = f; } // @S3
}

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/kotlin/controls/boundaries.kt
class Router {
    private var stored: () -> Unit = {}
    private var secondary: () -> Unit = {}
    val callback: () -> Unit
        get() = stored
    val other: () -> Unit
        get() = secondary
    fun install(f: () -> Unit, g: () -> Unit) {
        stored = f // @S1
        secondary = g // @S2
    }
    fun fire() { callback() } // @I1
    fun fireOther() { other() } // @I2
}
